### PR TITLE
fix case when aggregate buckets is undefined/null

### PR DESCRIPTION
**Pull request by jonathaningram**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -564,7 +564,7 @@ class QueryPipeline extends EventEmitter {
           return {
             type: "count",
             key: "buckets",
-            value: Object.values(aggregate.buckets.buckets).reduce(
+            value: Object.values(aggregate.buckets.buckets ?? {}).reduce(
```
Comment (benhinchley) [inline src/index.ts:567]: Is it possible for `aggregate.buckets` to be undefined or null??

Comment (jonathaningram) [reply src/index.ts:567]: It seems so?

image

Raw response:

```json
{"searchResponse":{"totalResults":"12758","time":"0.005802s","aggregates":{"count":{"buckets":{}}},"aggregateFilters":{"count":{"buckets":{}}},"results":[{"values":{"_id":{"single":"fd149182-4e15-5da8-f2ef-9478333df90c"},"bestSellingRank":{"single":"34"},"brand":{"single":"SanDisk"},"categories":{"repeated":{"values":["Cameras \u0026 Camcorders","Memory Cards","All Memory Cards"]}},"customerReviewCount":{"single":"273"},"description":{"single":"Save your photos and videos on this SanDisk Ultra Plus SDSDUSC-016G-AN6IN SDHC memory card, which features 16GB of storage space to hold plenty of your content. The rugged design resists damage from water, X-rays and more."},"free_shipping":{"single":"true"},"id":{"single":"3142535"},"image":{"single":"https://cdn.sajari.net/demos/bestbuy/images/3142535_sb.jpg"},"imageTags":{"repeated":{"values":["Memory card","Flash memory","Technology","Electronic device","Computer data storage","Electronics accessory"]}},"level1":{"single":"Cameras \u0026 Camcorders"},"level2":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards"},"level3":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards \u003e All Memory Cards"},"name":{"single":"SanDisk - Ultra Plus 16GB SDHC Class 10 UHS-1 Memory Card - Black/Gray/Red"},"popularity":{"single":"21225"},"price":{"single":"34.99"},"price_range":{"single":"1 - 50"},"rating":{"single":"4"},"type":{"single":"Secure digital memor"},"url":{"single":"https://www.bestbuy.com/site//3142535.p?cmp=SAJARI\u0026skuId=3142535\u0026utm_medium=search\u0026utm_campaign=sajari"}},"score":0.9948300000000001,"indexScore":1},{"values":{"_id":{"single":"587b012d-62ef-96f9-99a8-a206933c8210"},"bestSellingRank":{"single":"52"},"brand":{"single":"Apple"},"categories":{"repeated":{"values":["Audio","Headphones","All Headphones"]}},"customerReviewCount":{"single":"2524"},"description":{"single":"Control media playback, adjust the volume and answer and end calls on your Apple\u0026#174; iPod\u0026#174;, iPhone\u0026#174; or iPad\u0026#174; with these Apple EarPods\u0026#8482; MD827LL/A, which feature an in-line remote and mic for ease of use."},"free_shipping":{"single":"true"},"id":{"single":"6848136"},"image":{"single":"https://cdn.sajari.net/demos/bestbuy/images/6848136_sb.jpg"},"imageTags":{"repeated":{"values":["Headphones","Audio equipment","Technology","Electronic device"]}},"level1":{"single":"Audio"},"level2":{"single":"Audio \u003e Headphones"},"level3":{"single":"Audio \u003e Headphones \u003e All Headphones"},"name":{"single":"Apple - EarPods™ with Remote and Mic - White"},"popularity":{"single":"21462"},"price":{"single":"29.99"},"price_range":{"single":"1 - 50"},"rating":{"single":"4"},"type":{"single":"Earbud headphones"},"url":{"single":"https://www.bestbuy.com/site//6848136.p?cmp=SAJARI\u0026skuId=6848136\u0026utm_medium=search\u0026utm_campaign=sajari"}},"score":0.9947400000000001,"indexScore":1},{"values":{"_id":{"single":"f412c192-1134-4e2f-db19-5efc321f7575"},"bestSellingRank":{"single":"53"},"brand":{"single":"HP"},"categories":{"repeated":{"values":["Office \u0026 School Supplies","Printer Ink \u0026 Toner","Printer Ink"]}},"customerReviewCount":{"single":"1550"},"description":{"single":"This ink cartridge is compatible with select HP Deskjet, PhotoSmart and OfficeJet printers, and allows you to print documents with crisp, black text."},"free_shipping":{"single":"true"},"id":{"single":"1982034"},"image":{"single":"https://cdn.sajari.net/demos/bestbuy/images/1982034_sb.jpg"},"imageTags":{"repeated":{"values":["Liquid"]}},"level1":{"single":"Office \u0026 School Supplies"},"level2":{"single":"Office \u0026 School Supplies \u003e Printer Ink \u0026 Toner"},"level3":{"single":"Office \u0026 School Supplies \u003e Printer Ink \u0026 Toner \u003e Printer Ink"},"name":{"single":"HP - 564XL High-Yield Ink Cartridge - Black"},"popularity":{"single":"21421"},"price":{"single":"26.99"},"price_range":{"single":"1 - 50"},"rating":{"single":"4"},"type":{"single":"Ink cartridges"},"url":{"single":"https://www.bestbuy.com/site//1982034.p?cmp=SAJARI\u0026skuId=1982034\u0026utm_medium=search\u0026utm_campaign=sajari"}},"score":0.994735,"indexScore":1},{"values":{"_id":{"single":"b5990294-d9e8-5e5a-f40f-d044b89407d2"},"bestSellingRank":{"single":"58"},"brand":{"single":"SanDisk"},"categories":{"repeated":{"values":["Cameras \u0026 Camcorders","Memory Cards","All Memory Cards"]}},"customerReviewCount":{"single":"192"},"description":{"single":"With a 32GB storage capacity, this SanDisk Ultra Plus SDSDUSC-032G-AN6IN SDHC memory card features ample space to store your photos, high-definition videos and other content."},"free_shipping":{"single":"true"},"id":{"single":"3142553"},"image":{"single":"https://cdn.sajari.net/demos/bestbuy/images/3142553_sb.jpg"},"imageTags":{"repeated":{"values":["Memory card","Flash memory","Technology","Electronic device","Computer data storage","Electronics accessory"]}},"level1":{"single":"Cameras \u0026 Camcorders"},"level2":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards"},"level3":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards \u003e All Memory Cards"},"name":{"single":"SanDisk - Ultra Plus 32GB SDHC Class 10 UHS-1 Memory Card - Black/Gray/Red"},"popularity":{"single":"21430"},"price":{"single":"59.99"},"price_range":{"single":"50 - 100"},"rating":{"single":"4"},"type":{"single":"Secure digital memor"},"url":{"single":"https://www.bestbuy.com/site//3142553.p?cmp=SAJARI\u0026skuId=3142553\u0026utm_medium=search\u0026utm_campaign=sajari"}},"score":0.9947100000000001,"indexScore":1},{"values":{"_id":{"single":"33e6251e-8286-896e-b194-ce762129660f"},"bestSellingRank":{"single":"63"},"brand":{"single":"SanDisk"},"categories":{"repeated":{"values":["Cameras \u0026 Camcorders","Memory Cards","All Memory Cards"]}},"customerReviewCount":{"single":"193"},"description":{"single":"Expand the storage of your compatible device with this SanDisk Ultra Plus microSDHC memory card, which features a 32GB capacity to provide ample space for your media collection. The Class 10 UHS-1 speed enables high-fidelity video recording."},"free_shipping":{"single":"true"},"id":{"single":"3142635"},"image":{"single":"https://cdn.sajari.net/demos/bestbuy/images/3142635_sb.jpg"},"imageTags":{"repeated":{"values":["Memory card","Flash memory","Technology","Electronic device","Computer data storage","Font","Electronics accessory"]}},"level1":{"single":"Cameras \u0026 Camcorders"},"level2":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards"},"level3":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards \u003e All Memory Cards"},"name":{"single":"SanDisk - Ultra Plus 32GB microSDHC Class 10 UHS-1 Memory Card - Gray/Red"},"popularity":{"single":"21441"},"price":{"single":"59.99"},"price_range":{"single":"50 - 100"},"rating":{"single":"4"},"type":{"single":"Mobile phone media"},"url":{"single":"https://www.bestbuy.com/site//3142635.p?cmp=SAJARI\u0026skuId=3142635\u0026utm_medium=search\u0026utm_campaign=sajari"}},"score":0.994685,"indexScore":1},{"values":{"_id":{"single":"4f0de8dc-5b51-cd8a-7ac7-51a12c0ecf9a"},"bestSellingRank":{"single":"73"},"brand":{"single":"SanDisk"},"categories":{"repeated":{"values":["Cameras \u0026 Camcorders","Memory Cards","All Memory Cards"]}},"customerReviewCount":{"single":"140"},"description":{"single":"This SanDisk Ultra Plus 16GB microSDHC memory card features a Class 10 UHS-1 speed rating to ensure quality video recording performance, while the rugged construction protects your media from damage due to temperature, X-rays, shock and water."},"free_shipping":{"single":"true"},"id":{"single":"3142599"},"image":{"single":"https://cdn.sajari.net/demos/bestbuy/images/3142599_sb.jpg"},"imageTags":{"repeated":{"values":["Memory card","Flash memory","Technology","Computer data storage","Electronic device","Font","Electronics accessory"]}},"level1":{"single":"Cameras \u0026 Camcorders"},"level2":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards"},"level3":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards \u003e All Memory Cards"},"name":{"single":"SanDisk - Ultra Plus 16GB microSDHC Class 10 UHS-1 Memory Card - Gray/Red"},"popularity":{"single":"21268"},"price":{"single":"34.99"},"price_range":{"single":"1 - 50"},"rating":{"single":"4"},"type":{"single":"Mobile phone media"},"url":{"single":"https://www.bestbuy.com/site//3142599.p?cmp=SAJARI\u0026skuId=3142599\u0026utm_medium=search\u0026utm_campaign=sajari"}},"score":0.994635,"indexScore":1},{"values":{"_id":{"single":"4b45b778-c3a1-686b-94cd-1f676741833b"},"bestSellingRank":{"single":"78"},"brand":{"single":"SanDisk"},"categories":{"repeated":{"values":["Cameras \u0026 Camcorders","Memory Cards","All Memory Cards"]}},"customerReviewCount":{"single":"1543"},"description":{"single":"Easily store or transfer photos, videos and more with this SanDisk SDSDU-008G-A46 Ultra SDHC UHS-I Class 10 memory card that features an 8GB capacity and a Class 10 rating for fast read and write speeds."},"free_shipping":{"single":"true"},"id":{"single":"5577795"},"image":{"single":"https://cdn.sajari.net/demos/bestbuy/images/5577795_sb.jpg"},"imageTags":{"repeated":{"values":["Memory card","Flash memory","Technology","Electronic device","Computer data storage"]}},"level1":{"single":"Cameras \u0026 Camcorders"},"level2":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards"},"level3":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards \u003e All Memory Cards"},"name":{"single":"SanDisk - Ultra 8GB SDHC UHS-I Class 10 Memory Card - Black/Gray/Red"},"popularity":{"single":"20796"},"price":{"single":"24.99"},"price_range":{"single":"1 - 50"},"rating":{"single":"4"},"type":{"single":"Secure digital memor"},"url":{"single":"https://www.bestbuy.com/site//5577795.p?cmp=SAJARI\u0026skuId=5577795\u0026utm_medium=search\u0026utm_campaign=sajari"}},"score":0.9946100000000001,"indexScore":1},{"values":{"_id":{"single":"963256aa-3de3-de00-e5d4-23fc267294a9"},"bestSellingRank":{"single":"115"},"brand":{"single":"SanDisk"},"categories":{"repeated":{"values":["Cameras \u0026 Camcorders","Memory Cards","Micro SD (SD, SDHC, SDXC)","Micro SDHC"]}},"customerReviewCount":{"single":"271"},"description":{"single":"Save a variety of digital photos and 3D or full 1080p HD video on this SanDisk Extreme PLUS microSDHC UHS-I class U-1 memory card, which features a 32GB capacity to ensure plenty of storage space for important data."},"free_shipping":{"single":"true"},"id":{"single":"9387066"},"image":{"single":"https://cdn.sajari.net/demos/bestbuy/images/9387066_sb.jpg"},"imageTags":{"repeated":{"values":["Memory card","Flash memory","Font","Technology","Computer data storage","Electronic device"]}},"level1":{"single":"Cameras \u0026 Camcorders"},"level2":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards"},"level3":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards \u003e Micro SD (SD, SDHC, SDXC)"},"level4":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards \u003e Micro SD (SD, SDHC, SDXC) \u003e Micro SDHC"},"name":{"single":"SanDisk - Extreme PLUS 32GB microSDHC UHS-3 Memory Card - Red/Gold"},"popularity":{"single":"21359"},"price":{"single":"79.99"},"price_range":{"single":"50 - 100"},"rating":{"single":"4"},"type":{"single":"Mobile phone media"},"url":{"single":"https://www.bestbuy.com/site//9387066.p?cmp=SAJARI\u0026skuId=9387066\u0026utm_medium=search\u0026utm_campaign=sajari"}},"score":0.9944250000000001,"indexScore":1},{"values":{"_id":{"single":"254f3d4d-edf5-8002-34c6-098fd85510c0"},"bestSellingRank":{"single":"122"},"brand":{"single":"SanDisk"},"categories":{"repeated":{"values":["Cameras \u0026 Camcorders","Memory Cards","All Memory Cards"]}},"customerReviewCount":{"single":"462"},"description":{"single":"Store media with ease using this SanDisk 32GB Extreme PLUS SDHC memory card that features a spacious 32GB capacity and up to 90MB/sec. read speeds and up to 60MB/sec. write speeds for fast data transfer."},"free_shipping":{"single":"true"},"id":{"single":"3990822"},"image":{"single":"https://cdn.sajari.net/demos/bestbuy/images/3990822_sb.jpg"},"imageTags":{"repeated":{"values":["Memory card","Flash memory","Technology","Electronic device","Computer data storage"]}},"level1":{"single":"Cameras \u0026 Camcorders"},"level2":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards"},"level3":{"single":"Cameras \u0026 Camcorders \u003e Memory Cards \u003e All Memory Cards"},"name":{"single":"SanDisk - Extreme PLUS 32GB SDHC UHS-I Memory Card - Black"},"popularity":{"single":"21257"},"price":{"single":"79.99"},"price_range":{"single":"50 - 100"},"rating":{"single":"4"},"type":{"single":"Secure digital memor"},"url":{"single":"https://www.bestbuy.com/site//3990822.p?cmp=SAJARI\u0026skuId=3990822\u0026utm_medium=search\u0026utm_campaign=sajari"}},"score":0.9943900000000001,"indexScore":1},{"values":{"_id":{"single":"15096655-c642-146e-fe4d-a43cc5012473"},"bestSellingRank":{"single":"124"},"brand":{"single":"HP"},"categories":{"repeated":{"values":["Office \u0026 School Supplies","Printer Ink \u0026 Toner","Printer Ink"]}},"customerReviewCount":{"single":"1404"},"description":{"single":"Print your important documents and favorite photos with this dark black ink, which dries instantly and resists smudging"},"free_shipping":{"single":"true"},"id":{"single":"8891853"},"image":{"single":"https://cdn.sajari.net/demos/bestbuy/images/8891853_sb.jpg"},"imageTags":{"repeated":{"values":["Product","Technology","Electronic device","Ink","Printer consumable"]}},"level1":{"single":"Office \u0026 School Supplies"},"level2":{"single":"Office \u0026 School Supplies \u003e Printer Ink \u0026 Toner"},"level3":{"single":"Office \u0026 School Supplies \u003e Printer Ink \u0026 Toner \u003e Printer Ink"},"name":{"single":"HP - 564 Ink Cartridge - Black"},"popularity":{"single":"21346"},"price":{"single":"15.99"},"price_range":{"single":"1 - 50"},"rating":{"single":"4"},"type":{"single":"Ink cartridges"},"url":{"single":"https://www.bestbuy.com/site//8891853.p?cmp=SAJARI\u0026skuId=8891853\u0026utm_medium=search\u0026utm_campaign=sajari"}},"score":0.99438,"indexScore":1}]}}
```

Comment (benhinchley) [reply src/index.ts:567]: What was the request sent??

Comment (jonathaningram) [reply src/index.ts:567]: For posterity:

```
curl 'http://jsonapi-us-valkyrie.sajari.net/sajari.api.pipeline.v1.Query/Search' \
  -H 'Connection: keep-alive' \
  -H 'Accept: application/json' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36' \
  -H 'Content-Type: text/plain' \
  -H 'Origin: http://local.sajari.com:3000' \
  -H 'Referer: http://local.sajari.com:3000/console/collection/preview' \
  -H 'Accept-Language: en-GB,en-US;q=0.9,en;q=0.8' \
  --data-binary $'{"metadata":{"project":["redacted"],"collection":["redacted"],"user-agent":["sdk-js-2.0.0-alpha.10"]},"request":{"pipeline":{"name":"query","version":"1"},"tracking":{"type":"NONE"},"values":{"q":"test","filter":"_id \u0021= \'\'","fields":"","page":"1"}}}' \
  --compressed \
  --insecure
```

**Pull request merged**